### PR TITLE
feat: Allow user to pass null (empty string) for ipv6_ipam_pool_id

### PR DESCRIPTION
**Pull request by nshenry03**
## Description

This change allows users to call the module from their own module while passing `null` or an empty string (`""`) to this module for the `ipv6_ipam_pool_id` variable.

## Motivation and Context

We have our own wrapper module for VPCs that calls this module. We are using [Terragrunt](https://terragrunt.gruntwork.io/) to keep our code DRY.

In our `_envcommon` for VPCs, we have

```
locals {
  ...

   # Automatically load environment-level variables
  environment_vars = read_terragrunt_config(find_in_parent_folders("env.hcl"))

  # Extract out common variables for reuse
  ipv6_ipam_pool_id = try(local.environment_vars.locals.vpc_ipv6_ipam_pool_id, null)
  use_ipam_pool       = local.ipv6_ipam_pool_id != null

  ...
}

inputs = {
  ...

  use_ipam_pool     = local.use_ipam_pool
  ipv6_ipam_pool_id = local.ipv6_ipam_pool_id

  ...
}
```

and we're getting the following error when `ipv6_ipam_pool_id` is `null`:

```
Error: Conflicting configuration arguments

  on .terraform/modules/vpc/main.tf line 27, in resource "aws_vpc" "this":
  27:   assign_generated_ipv6_cidr_block = var.enable_ipv6 && !var.use_ipam_pool  true : null

"assign_generated_ipv6_cidr_block": conflicts with ipv6_ipam_pool_id

Error: Conflicting configuration arguments

  on .terraform/modules/vpc/main.tf line 29, in resource "aws_vpc" "this":
  29:   ipv6_ipam_pool_id                = var.ipv6_ipam_pool_id

"ipv6_ipam_pool_id": conflicts with assign_generated_ipv6_cidr_block

time=2023-03-22T18:46:38Z level=error msg=Terraform invocation failed in /tmp/274f71cb-cc98-48bf-bfe2-5b91e9064b0a/sase-preprod/us-west-2/sase-preprod/vpc/.terragrunt-cache/7bSSPeYRzQyBwlya0Kvd1mmg3ls/m48TkmguM93MPX0LUY4qqcafT_g prefix=[/tmp/274f71cb-cc98-48bf-bfe2-5b91e9064b0a/sase-preprod/us-west-2/sase-preprod/vpc] 
time=2023-03-22T18:46:38Z level=error msg=1 error occurred:
	* exit status 1
```

This relates to https://github.com/hashicorp/terraform/issues/24142

So this is a fix for us (and others), which allows `null` or empty strings to be passed to this module.

## Breaking Changes

No, it works the same as before.

## How Has This Been Tested?
- [x] I have updated at least one of the `examples/*` to demonstrate and validate my change(s)
- [x] I have tested and validated these changes using one or more of the provided `examples/*` projects
- [x] I have executed `pre-commit run -a` on my pull request

```
hnicholas7DYYW:ipv6 hnicholas$ pre-commit run
Terraform fmt............................................................Passed
Terraform validate.......................................................Passed
Terraform docs...........................................................Passed
Terraform validate with tflint...........................................Passed
check for merge conflicts................................................Passed
fix end of files.........................................................Passed
```

### CODE REVIEW

File: main.tf [hunk 1]
```diff
@@ -35,7 +35,7 @@ resource "aws_vpc" "this" {
 
   assign_generated_ipv6_cidr_block     = var.enable_ipv6 && !var.use_ipam_pool ? true : null
   ipv6_cidr_block                      = var.ipv6_cidr
-  ipv6_ipam_pool_id                    = var.ipv6_ipam_pool_id
+  ipv6_ipam_pool_id                    = var.ipv6_ipam_pool_id != "" ? var.ipv6_ipam_pool_id : null
```
Comment (bryantbiggs) [inline main.tf:38]: > So this is a fix for us (and others), which allows null or empty strings to be passed to this module.

It already supports this, and it defaults to `null` - I'm not following why this change is warranted or what problem it is addressing

Comment (nshenry03) [reply main.tf:38]: Hey @bryantbiggs ,

It's working around this issue: https://github.com/hashicorp/terraform/issues/24142

I'm seeing this error when passing `ipv6_ipam_pool_id  = null` (as `null` and not `"null"`):

```
Error: Conflicting configuration arguments

  on .terraform/modules/vpc/main.tf line 27, in resource "aws_vpc" "this":
  27:   assign_generated_ipv6_cidr_block = var.enable_ipv6 && !var.use_ipam_pool  true : null

"assign_generated_ipv6_cidr_block": conflicts with ipv6_ipam_pool_id

Error: Conflicting configuration arguments

  on .terraform/modules/vpc/main.tf line 29, in resource "aws_vpc" "this":
  29:   ipv6_ipam_pool_id                = var.ipv6_ipam_pool_id

"ipv6_ipam_pool_id": conflicts with assign_generated_ipv6_cidr_block

time=2023-03-22T18:46:38Z level=error msg=Terraform invocation failed in /tmp/274f71cb-cc98-48bf-bfe2-5b91e9064b0a/sase-preprod/us-west-2/sase-preprod/vpc/.terragrunt-cache/7bSSPeYRzQyBwlya0Kvd1mmg3ls/m48TkmguM93MPX0LUY4qqcafT_g prefix=[/tmp/274f71cb-cc98-48bf-bfe2-5b91e9064b0a/sase-preprod/us-west-2/sase-preprod/vpc] 
time=2023-03-22T18:46:38Z level=error msg=1 error occurred:
	* exit status 1
```

Comment (bryantbiggs) [reply main.tf:38]: why do you need to re-pass `null` when it defaults to `null`?

Comment (nshenry03) [reply main.tf:38]: Because we're using Terragrunt to create VPCs in 5 accounts and 27 different regions...  Of those, some of those were created before we added support for BYOIPv6 so we use logic:

```
locals {
  ..

  #############################################################################
  # Extract out common variables for reuse
  #############################################################################
  ...
  ipv6_ipam_pool_id     = try(local.environment_vars.locals.vpc_ipv6_ipam_pool_id, null)
  use_ipam_pool         = local.ipv6_ipam_pool_id != null
  ...
}


inputs = {
  ...
  ipv6_ipam_pool_id                = local.ipv6_ipam_pool_id
  use_ipam_pool                    = local.use_ipam_pool
  ..
```
